src/reducers/sources.js

```diff
@@ -284,8 +284,12 @@ export function getSourceByURL(state: OuterState, url: string) {
   return state.sources.sources.find(source => source.get("url") === url);
 }
 
+export function getSourceInSources(sources: I.Map<string, Source>, id: string) {
```

Comment:
Yeah, not sure what to do without finding a better name for either the top-level "sources state" or the nested "sources by id map".